src/Mod/Fem/Gui/TaskFemConstraintOnBoundary.cpp

```diff
@@ -0,0 +1,137 @@
+ *   Authors: Ajinkya Dahale                                               *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "PreCompiled.h"
+
+#ifndef _PreComp_
+# include <BRepAdaptor_Curve.hxx>
+# include <BRepAdaptor_Surface.hxx>
+# include <Geom_Line.hxx>
+# include <Geom_Plane.hxx>
+# include <Precision.hxx>
+
+# include <QAction>
+# include <QKeyEvent>
+# include <QMessageBox>
+# include <QRegExp>
+# include <QTextStream>
+
+# include <TopoDS.hxx>
+# include <gp_Ax1.hxx>
+# include <gp_Lin.hxx>
+# include <gp_Pln.hxx>
+# include <sstream>
+#endif
+
+#include "TaskFemConstraintOnBoundary.h"
+#include <App/Application.h>
+#include <Base/Tools.h>
+#include <Gui/Command.h>
+#include <Gui/Selection.h>
+#include <Gui/SelectionFilter.h>
+// #include <Mod/Part/App/PartFeature.h>
+// #include <Mod/PartDesign/Gui/ReferenceSelection.h>
+
+
+using namespace FemGui;
+using namespace Gui;
+
+/* TRANSLATOR FemGui::TaskFemConstraintOnBoundary */
+
+TaskFemConstraintOnBoundary::TaskFemConstraintOnBoundary(ViewProviderFemConstraint *ConstraintView, QWidget *parent, const char* pixmapname)
+    : TaskFemConstraint(ConstraintView, parent, pixmapname)
+{
+}
+
+TaskFemConstraintOnBoundary::~TaskFemConstraintOnBoundary()
+{
+}
+
+void TaskFemConstraintOnBoundary::_addToSelection(bool checked)
+{
+    if (checked)
+    {
+        const auto& selection = Gui::Selection().getSelectionEx(); //gets vector of selected objects of active document
+        if (selection.empty()) {
+            this->clearButtons(refAdd);
+            selChangeMode = refAdd;
+            // TODO: set up depending on selChangeMode
```

Comment:
Do you plan this as separate PR? (a ToDo in new files is somehow unusual ;-) )

Reply:
This particular one was not necessary, I believe. It was just to make sure add and remove did different things when I used the same template for both.

Reply:
In general I leave some TODO's if I believe there are places where improvements can be made but I don't have the time or any other reason.